spark/src/main/java/org/apache/iceberg/actions/BaseAction.java

```diff
@@ -94,9 +102,30 @@ protected String metadataTableName(String tableName, MetadataTableType type) {
     return buildValidDataFileDF(spark, table().toString());
   }
 
+  private static class ReadManifest implements FlatMapFunction<ManifestFileBean, String> {
```

Comment:
In other actions, we have private static methods at the end of the classes like below:

```
private static FlatMapFunction<Iterator<String>, String> listDirsRecursively(
      Broadcast<SerializableConfiguration> conf,
      long olderThanTimestamp) {

    return (FlatMapFunction<Iterator<String>, String>) dirs -> {
      // logic
    };
  }
```

I am OK with this approach too but may make sense to align it with other places.

Reply:
Moved.